Add Tale licenses

ThomasThelen opened the pull request:
This PR has two parts, which are separated into respective commits. 

The first adds a new field to the Tale, `license`. 
The second adds an endpoint that returns a list of supported licenses. I put three fields for each record:
   1. The common name
   2. The spdx
   3. A compliant portion of the license needed to properly attribute a work

During the dev call about this feature, we decided that a URL would be the best approach however, I wasn't able to find a suitable endpoint that returned the text of a given license, and it seems that data repositories (Dataverse and DataONE) supply legal subsets of the license along with the URL. The text was taken from packages on DataONE however, similar text can be found across other repositories.

To Test Licenses:
   1. Check this branch out
   2. Deploy
   3. Visit the Girder API page
   4. Call Get on the license endpoint

To Test Tale Licenses:
   1. Checkout this branch
   2. Deploy
   3. Create a new Tale
   4. Query the Tale from Girder
   5. See that it has a default license applied

craig-willis commented on server/rest/license.py:
```diff
@@ -0,0 +1,36 @@
+    )
+    def get_licenses(self, params):
+        return [
+            {
```
At some point, should we consider moving these into the database?  For now I think this is fine.

Xarthisius replied:
I'd suggest a different approach for now:
1. Wrap it in a class (`WholeTaleLicenses`?). Don't worry about mutability and let that array be an attribute of it (I don't have a good name for it... `licenses`).
2. In this method you'd just return `WholTaleLicenses.licenses`
3. Add a method to the class that would return supported *spdx*es. I think it'll be useful in couple of places (tests, validation etc.) e.g.:
  ```
  def supported_spdxes(self):
      return {license['spdx'] for license in self.licenses}
  ```

craig-willis commented on plugin_tests/license_test.py:
```diff
@@ -0,0 +1,38 @@
+    base.stopServer()
+
+
+class RecipeTestCase(base.TestCase):
```
Should be LicenseTestCase

craig-willis commented on plugin_tests/tale_test.py:
```diff
@@ -546,6 +549,7 @@ def testTaleUpdate(self):
         published = True
         doi = 'doi:10.x.zz'
         published_uri = 'atestURI'
+        license = 'CC-BY'
```
This isn't a valid license and you're not doing validation in Tale creation.

craig-willis commented on server/rest/license.py:
```diff
@@ -0,0 +1,36 @@
+        return [
+            {
+                'name': 'Creative Commons Zero v1.0 Universal',
+                'spdx': 'CC0-1.0',
```
This is really the spdx license identifier? You also call it 'license' in the Tale object when it is just a reference to the spdc license identifier.  So there's an inconsistency?   Maybe call it licenseSpdxId or licenseId throughout ?

ThomasThelen commented:
## Changes


### Moved License to a Class in lib/license.py
I added `default_spdx()` which wasn't in the comments above to get rid of the magic default license strings that I had used.

### Renamed tale['license'] to tale['licenseSPDX']
A complete list of SPDXes can be found here https://spdx.org/licenses/

### Validated Tale License
I added a check in the Tale model to make sure any license is in our list of supported licenses.
